components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1154,6 +1189,44 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
     }
 
+    async teamCheckout(ctx: TraceContext, teamId: string, planId: string): Promise<{}> {
+        traceAPIParams(ctx, { teamId, planId });
+
+        const user = this.checkUser("teamCheckout");
+
+        const team = await this.teamDB.findTeamById(teamId);
+        if (!team) {
+            throw new ResponseError(ErrorCodes.NOT_FOUND, "Team not found");
+        }
+        const members = await this.teamDB.findMembersByTeam(team.id);
+        await this.guardAccess({ kind: "team", subject: team, members }, "update");
+
+        const coupon = await this.findAvailableCouponForPlan(user, planId);
+
+        const email = User.getPrimaryEmail(user);
```

Comment:
😢  
at least this is not used to id the customers. 
just the regular note: `User.getPrimaryEmail` is not stable.